### PR TITLE
Support building zephyr with upstream open-amp and libmetal

### PR DESCRIPTION
**Pull request by wmamills**
Prototype stage only.

TODO:
DONE: modify to work on Windows also
DONE: Get rid of wam remote stuff
WIP will be a different PR: Add CI github action workflow

### CODE REVIEW

**Comment by wmamills** [comment]
I have updated the *-module branches with a new commit that remove the use of the symlink.
That is hard to show here in the PR but here is a pointer:
https://github.com/wmamills/openamp-system-reference/commits/open-amp-module

Please note the commented lines to find the libmetal headers and libs.
I believe that is redundant with what Zephyr cmake already does for every module.
It works when I did not update them to the new paths and it works when I commented them out.

**Review by arnopo — COMMENTED** [review]
I tested with success, I also proposed an alternative to squash the libmetal-module and open-amp-module in one

**Comment by uLipe** [comment]
Just adding to the conversation, I also tested with no effort, it worked very well.

File: west.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ manifest:
       url-base: https://github.com/OpenAMP
     - name: zephyr
       url-base: https://github.com/zyphyrproject-rtos
+    - name: wam
+      url-base: https://github.com/wmamills
```
Comment (arnopo) [inline west.yml:11]: could you update to rely on https://github.com/OpenAMP/openamp-zephyr-modules?
I know that there is a way to specify the revision as a PR reference, waiting merge of https://github.com/OpenAMP/openamp-zephyr-modules/pull/1, but i can ind the exact syntax for now.

**Comment by wmamills** [comment]
This is now clean and ready.  This is no longer an RFC.
@arnopo please take a look

**Review by arnopo — APPROVED** [review]

**Comment by wmamills** [comment]
@uLipe any comment?

**Comment by uLipe** [comment]
@wmamill, no, still LGTM.

**Review by uLipe — APPROVED** [review]

**Pull request merged**